src/storage/mongodb/TenantStorage.ts

```diff
@@ -48,7 +48,10 @@ export default class TenantStorage {
     } else {
       tenantFilter._id = new ObjectId();
     }
-    DatabaseUtils.addLastChangedCreatedProps(tenantMDB, tenantToSave);
+    // Remove ID
+    delete tenantMDB.id;
+    // Add Last Changed/Created props
+    DatabaseUtils.addLastChangedCreatedProps(tenantMDB, Utils.cloneObject(tenantToSave));
```

Comment:
Because Validators do not return a new cleared objet but mutates the original data. So in this case, the variable `tenantMDB` and `tenantToSave` are referencing the same object.

In the `addLastChangedCreatedProps` method, the `createdBy` and `lastChangedBy` attributes of the first parameter are set to null to be replaced by the second parameter attributes values. However, as they both reference the same object, it results with a null value at the end

Reply:
You can remove the 'null' assignment and move it a new 'else' statement in method addLastChangedCreatedProps().
What do you think?

Reply:
Actually using ternary conditions in variable assignment fixes the problem and simplify the code